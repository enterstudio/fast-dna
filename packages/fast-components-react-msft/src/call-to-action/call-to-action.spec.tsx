import * as React from "react";
import * as ShallowRenderer from "react-test-renderer/shallow";
import * as Adapter from "enzyme-adapter-react-16/build";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import CallToAction,
{
    CallToActionAppearance,
    CallToActionProps,
    ICallToActionHandledProps,
    ICallToActionManagedClasses,
    ICallToActionUnhandledProps
} from "./call-to-action";
import { ICallToActionClassNameContract } from "@microsoft/fast-components-class-name-contracts-msft";

/*
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("call to action snapshot", (): void => {
    generateSnapshots(examples);
});

// describe("call to action", (): void => {
//     const Component: React.ComponentClass<ICallToActionHandledProps & ICallToActionManagedClasses> = examples.component;
//     const managedClasses: ICallToActionClassNameContract = {
//         hypertext: "hypertext",
//         callToAction_primary: "call-to-action-primary",
//         callToAction_secondary: "call-to-action-secondary",
//         callToAction_lightweight: "call-to-action-lightweight",
//         callToAction_span: "call-to-action-span",
//         "@keyframes cta-glyph-move-out": "cta-glyph-move-out",
//         "@keyframes cta-glyph-move-in": "cta-glyph-move-in",
//         "@keyframes cta-text-move-in": "cta-text-move-in",
//         "@keyframes cta-text-move-out": "cta-text-move-out"
//     };
//     const href: string = "#";

//     test("should implement unhandledProps", () => {
//         const handledProps: ICallToActionProps & ICallToActionManagedClasses = {
//             managedClasses,
//             href,
//             children: "text"
//         };

//         const unhandledProps: ICallToActionUnhandledProps = {
//             "data-my-custom-attribute": true
//         };

//         const props: CallToActionProps = {...handledProps, ...unhandledProps};

//         const rendered: any = shallow(
//             <Component {...props} />
//         );

//         expect(rendered.first().prop("data-my-custom-attribute")).toEqual(true);
//     });

//     // tslint:disable-next-line:max-line-length
//     test("should set a className that matches callToAction_primary managedClass when CallToActionAppearance.primary is passed to the appearance prop", () => {
//         const props: IButtonHandledProps = {
//             appearance: CallToActionAppearance.primary
//         };

//         const rendered: any = shallow(
//             <Component {...props}/>
//         );

//         const callToAction: any = rendered.first().shallow();
//         // Get the expected className value from the list of generated managed classes
//         const expectedClassName: string = callToAction.instance().props.managedClasses.callToAction_primary;

//         expect(callToAction.instance().props.appearance).toEqual(CallToActionAppearance.primary);
//         // Generated managedClass should be passed to className
//         expect(callToAction.prop("className")).toBe(expectedClassName);
//     });

//     // tslint:disable-next-line:max-line-length
//     test("should set a className that matches callToAction_secondary managedClass when CallToActionAppearance.secondary is passed to the appearance prop", () => {
//         const props: IButtonHandledProps = {
//             appearance: CallToActionAppearance.secondary
//         };

//         const rendered: any = shallow(
//             <Component {...props}/>
//         );

//         const callToAction: any = rendered.first().shallow();
//         // Get the expected className value from the list of generated managed classes
//         const expectedClassName: string = callToAction.instance().props.managedClasses.callToAction_secondary;

//         expect(callToAction.instance().props.appearance).toEqual(CallToActionAppearance.secondary);
//         // Generated managedClass should be passed to className
//         expect(callToAction.prop("className")).toBe(expectedClassName);
//     });

//     // tslint:disable-next-line:max-line-length
//     test("should set a className that matches callToAction_lightweight managedClass when CallToActionAppearance.lightweight is passed to the appearance prop", () => {
//         const props: IButtonHandledProps = {
//             appearance: CallToActionAppearance.lightweight
//         };

//         const rendered: any = shallow(
//             <Component {...props}/>
//         );

//         const callToAction: any = rendered.first().shallow();
//         // Get the expected className value from the list of generated managed classes
//         const expectedClassName: string = callToAction.instance().props.managedClasses.callToAction_lightweight;

//         expect(callToAction.instance().props.appearance).toEqual(CallToActionAppearance.lightweight);
//         // Generated managedClass should be passed to className
//         expect(callToAction.prop("className")).toBe(expectedClassName);
//     });
// });
