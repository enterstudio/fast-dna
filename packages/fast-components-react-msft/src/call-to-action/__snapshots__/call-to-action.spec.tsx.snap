// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`call to action snapshot Call to action: 0 1`] = `
<a
  className="button-0-1-16 button_primary-0-1-17 callToAction-0-1-5"
  data-sketch-symbol="Call to action - primary"
  href="https://www.microsoft.com/en-us/"
>
  <span
    className="button_textContainer-0-1-6"
  >
    Primary call to action
  </span>
  <div
    className="glyph-0-1-7"
    dangerouslySetInnerHTML={
      Object {
        "__html": "FOO",
      }
    }
  />
</a>
`;

exports[`call to action snapshot Call to action: 1 1`] = `
<a
  className="button-0-1-38 button_lightweight-0-1-41 callToAction-0-1-27"
  data-sketch-symbol="Call to action - lightweight"
  href="https://www.microsoft.com/en-us/"
>
  <span
    className="button_textContainer-0-1-43"
  >
    Lightweight call to action
    <div
      className="glyph-0-1-29"
      dangerouslySetInnerHTML={
        Object {
          "__html": "FOO",
        }
      }
    />
  </span>
</a>
`;

exports[`call to action snapshot Call to action: 2 1`] = `
<a
  className="button-0-1-60 button_justified-0-1-64 callToAction-0-1-49"
  data-sketch-symbol="Call to action - justified"
  href="https://www.microsoft.com/en-us/"
>
  <span
    className="button_textContainer-0-1-65"
  >
    Secondary call to action
    <div
      className="glyph-0-1-51"
      dangerouslySetInnerHTML={
        Object {
          "__html": "FOO",
        }
      }
    />
  </span>
</a>
`;
